podpac/core/node.py

```diff
@@ -139,8 +139,18 @@ def _validate_units(self, d):
 
     def __init__(self, **kwargs):
         """ Do not overwrite me """
+        # make tagged "readonly" and "attr" traits read_only
+        for name, trait in self.traits().items():
+            if trait.metadata.get("readonly") or trait.metadata.get("attr"):
+                trait.read_only = True
+
         tkwargs = self._first_init(**kwargs)
 
+        # set tagged "readonly" and "attr" traits using set_trait
+        for name, trait in self.traits().items():
+            if (trait.metadata.get("readonly") or trait.metadata.get("attr")) and name in tkwargs:
+                self.set_trait(name, tkwargs.pop(name))
+
         # Call traitlest constructor
```

Comment:
Simple implementation suggestion: 

```python
In [3]: class T(tl.HasTraits):
   ...:     a = tl.Int().tag(readonly=True)
   ...:     b = tl.Int()
   ...:     def __init__(self, **kwargs):
   ...:         super().__init__(**kwargs)
   ...:         for name, trait in self.traits().items():
   ...:             if (trait.metadata.get("readonly") and name in kwargs):
   ...:                 trait.read_only = True
   ...:

In [4]: t = T(a=1, b=2)

In [5]: t.a=3
---------------------------------------------------------------------------
TraitError                                Traceback (most recent call last)
<ipython-input-5-012fe921ffc2> in <module>
----> 1 t.a=3

C:\Anaconda3\envs\podpac\lib\site-packages\traitlets\traitlets.py in __set__(self, obj, value)
    581         """
    582         if self.read_only:
--> 583             raise TraitError('The "%s" trait is read-only.' % self.name)
    584         else:
    585             self.set(obj, value)

TraitError: The "a" trait is read-only.

In [6]: t.b=2
```

Reply:
Ah, yeah, the issue is that the read_only flag gets set on the *class* trait, so with this implementation it doesn't work the second time:

```
>>> t1 = T(a=1)
>>> t2 = T(a=1)
TraitError: The "a" trait is read-only
```

Reply:
I figured there was a reason! Wow, that's crazy. A comment in the code might be helpful...